src/Visualization/Python/GenerateTetrahedralConnectivity.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+from typing import Optional
+
+import click
+import h5py
+import numpy as np
+import rich
+import scipy.spatial as spatial
+
+from spectre.support.CliExceptions import RequiredChoiceError
+from spectre.support.Logging import configure_logging
+from spectre.Visualization.GenerateXdmf import available_subfiles
+
+logger = logging.getLogger(__name__)
+
+
+def generate_tetrahedral_connectivity(
+    h5files,
+    subfile_name: str,
+    relative_paths: bool = True,
+    start_time: Optional[float] = None,
+    stop_time: Optional[float] = None,
+    stride: int = 1,
+    coordinates: str = "InertialCoordinates",
+    delete_tetrahedral_connectivity: bool = False,
+):
+    """Generate tetrahedral connectivity using scipy.spatial.Delaunay
```

Comment:
Question: how does this handle shared grid points on LGL element boundaries? The tetrahedra connecting these points would be degenerate.

Reply:
I haven't read the paper that describes qhull, but as far as I can tell it works fine and doesn't produce any double connectivity.